dfastmi/gui/dialog_view.py

```diff
@@ -102,37 +105,49 @@ def update_reach(self, data):
         self.slength.setText(self.view_model.slength)
     
     def update_condition_files(self):
-        for condition_tab_index, reference_file in enumerate(self.view_model.reference_files):
-            tab = self._tabs.widget(condition_tab_index +1)
-            if tab:
-                prefix = str(condition_tab_index)+"_"
-                key = prefix +"file1"
-                input_textbox = tab.findChild(QtWidgets.QLineEdit, key)
-                if input_textbox:
-                    input_textbox.setText(reference_file)
+        for condition_discharge, reference_file in self.view_model.reference_files.items():
+            prefix = str(condition_discharge)+"_"
+            key = prefix +"file1"
+            input_textbox = self.general_widget.findChild(QtWidgets.QLineEdit, key)
+            if input_textbox:
+                input_textbox.setText(reference_file)
         
-        for condition_tab_index, measure_file in enumerate(self.view_model.measure_files):
-            tab = self._tabs.widget(condition_tab_index +1)
-            if tab:
-                prefix = str(condition_tab_index)+"_"
-                key = prefix +"file2"
-                input_textbox = tab.findChild(QtWidgets.QLineEdit, key)
-                if input_textbox:
-                    input_textbox.setText(measure_file)
+        for condition_discharge, measure_file in self.view_model.measure_files.items():        
+            prefix = str(condition_discharge)+"_"
+            key = prefix +"file2"
+            input_textbox = self.general_widget.findChild(QtWidgets.QLineEdit, key)
+            if input_textbox:
+                input_textbox.setText(measure_file)
     
     def update_qvalues_tabs(self):
-        hydro_q = self.view_model.current_reach.hydro_q
-        tabs = self._tabs
-        for j in range(tabs.count()-2,-1,-1):
-            tabs.removeTab(1+j)            
+        self._clear_conditions()
+        for discharge in self.view_model.current_reach.hydro_q:
+            prefix = str(discharge) + "_"
+            qval = str(discharge) + " m3/s"
+            self.add_condition_line(prefix, discharge, qval)
+
+    def _clear_conditions(self):
```

Comment:
I think it would be easier and more readable to just initialize a new empty `grid_layout`..

Reply:
I would love a new implementation.... but i don't know how... sorry! please help me :-)

Reply:
What are you exactly trying to do here, because it feels like you try to do something that should feels like it would be a standard funtion of widget. (that being said, I am no QT expert, so I have no actual footing on my claim)
Maybe we can look into making this method more readable and less indentable.

Reply:
In `create_qt_application` we setup a new grid layout (which is already empty). We can do the same here, instead of going through all of the columns/rows and clearing them. 
I can try to help with that if you like.

Reply:
Can we look at this in a new issue together? I can't get it as simple as you request

Reply:
DFAST-185